### PR TITLE
New sg_qicons, sg_qwidgets, message_box modules and improvements to delegates and filtering modules

### PR DESCRIPTION
**Pull request by staceyoue**
**Changes:**

ViewItemDelegate:
- Support drawing Check Box
- Render disabled buttons as greyed out
- Allow text to wrap instead of elide
- More general minor improvements to customize the view through the delegate, minor bug fixes

Filtering:
- Specify accepted fields only for filtering

**New:**
- **sg_qicons** module: Module includes icons to use throughout Toolkit, this module takes care of including the icon files and compiling the resources files for other Toolkit components to use by just importing the framework
- **sg_qwidgets** module: Module to create Qt widgets to help create and manage widget styles -- all Qt widgets have class name `SG{Qt_widget_name}` (e.g. SGQWidget). This way widgets can be easily styled using style.qss file.
- **messag_box** module: custom message box dialog to allow more customization than the out-of-the-box QMessageBox

### CODE REVIEW

**Review by 000paradox000 — CHANGES_REQUESTED** [review]
LGTM Please review my comment on icon file names

File: python/activity_stream/widget_note.py [hunk 1]
```diff
@@ -319,9 +319,7 @@ def set_selected(self, state):
         self.selection_changed.emit(self.selected, self._note_id)
 
     def _add_attachment_group(self, attachments, after_note):
-        """
-
-        """
+        """ """
```
[the remark this answers is not in the document]
Comment (staceyoue) [reply python/activity_stream/widget_note.py]: I didn't actually add this function here -- this is just the pre-commit formatting that has modified the file.

File: python/filtering/filter_definition.py [hunk 2]
```diff
@@ -112,6 +113,19 @@ def filter_roles(self):
     def filter_roles(self, value):
         self._filter_roles = value
 
+    @property
+    def accept_fields(self):
+        """
+        Get or set the fields that will are accepted when building the filter definition.
```
[the remark this answers is not in the document]
Comment (staceyoue) [reply python/filtering/filter_definition.py:119]: For properties in python, only the `getter` function's docstring is displayed, so it is suggested to indicate that this property can set the value, if there is a setter function defined for it.

For example, if you run `help(class)` you'll only see the `getter` docstrings that are defined (the setter docstrings are ignored)

File: python/filtering/filter_definition.py [hunk 3]
```diff
@@ -112,6 +113,19 @@ def filter_roles(self):
...
+
+        Set this list to None or empty to accept all fields.
```
[the remark this answers is not in the document]
Comment (staceyoue) [reply python/filtering/filter_definition.py:121]: How about:

>         Set the value of this property None or the empty list if the filter should accept all given fields (e.g. it does not ignore any fields).

**Review by 000paradox000 — APPROVED** [review]
LGTM thanks for the good job

**Review by 000paradox000 — COMMENTED** [review]
Very very good job. 

Just some comments:
- Method names could be more pythonic if possible
- There are many codecov warnings. Could be reduced easily?

**Pull request merged**